### PR TITLE
#189 fixed issue

### PR DESCRIPTION
**Pull request by xhoaluu**
@jaimecasero 

Please help me to review

### CODE REVIEW

File: sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java [hunk 1]
```diff
@@ -191,10 +191,10 @@ public void dispatchMessage(final SipProvider sipProvider, SipServletMessageImpl
 							return ;
 						} else {
 							if(poppedRouteHeader != null) {
-								throw new DispatcherException(Response.SERVER_INTERNAL_ERROR, "cannot find the application to handle this subsequent request " + request +
+								throw new DispatcherException(Response.CALL_OR_TRANSACTION_DOES_NOT_EXIST, "cannot find the application to handle this subsequent request " + request +
```
Comment (jaimecasero) [inline sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java]: good job at locating where to map the error code!!!

Comment (jaimecasero) [inline sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java]: Could you please add a comment refereing to the solved issue. Add the link to the issue, and brief explanation to why we are changing the error code. References to RFC are welcomed

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/concurrency/ConcurrentyControlSipSessionIsolationTest.java [hunk 2]
```diff
@@ -101,7 +101,7 @@ public void testElapsedTimeAndSessionOverlapping() throws InterruptedException,
 		SipURI fromAddress = senderProtocolObjects.addressFactory.createSipURI(
 				fromName, fromSipAddress);
 				
-		String toUser = "receiver";
+		String toUser = "receiverWithTag";
```
Comment (jaimecasero) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/concurrency/ConcurrentyControlSipSessionIsolationTest.java]: do  we neeed to modify this existing test?

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java [hunk 3]
```diff
@@ -844,6 +844,30 @@ public void testShootmeSendBye407Terminated() throws InterruptedException, SipEx
...
+        // This trick is for creating fake dialog at UA for sending sub sequence 
+        sender.sendSipRequest("INVITE", fromAddress, toAddress, null, null, false);
+        Thread.sleep(2000);
+        sender.sendInDialogSipRequest("INFO", "1", "text", "plain", null, null);
+        Thread.sleep(100);
+        assertTrue(sender.getFinalResponse().getStatusCode() == 481);
+        sender.sendBye();
+        Thread.sleep(100);
+        assertTrue(!sender.getOkToByeReceived());
+        assertTrue(sender.getFinalResponse().getStatusCode() == 481);
```
Comment (jaimecasero) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java]: i reckon this is to assert response to latest BYE is still 481

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java [hunk 4]
```diff
@@ -844,6 +844,30 @@ public void testShootmeSendBye407Terminated() throws InterruptedException, SipEx
...
+        SipURI toAddress = senderProtocolObjects.addressFactory.createSipURI(
+                toUser, toSipAddress);
+        
+        sender.setSendBye(false);
+        // This trick is for creating fake dialog at UA for sending sub sequence 
+        sender.sendSipRequest("INVITE", fromAddress, toAddress, null, null, false);
+        Thread.sleep(2000);
+        sender.sendInDialogSipRequest("INFO", "1", "text", "plain", null, null);
+        Thread.sleep(100);
+        assertTrue(sender.getFinalResponse().getStatusCode() == 481);
```
Comment (jaimecasero) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java]: what is this asserting? what is this response supposed to be? is this the response to the initial INVITE?

File: sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java (hunk 1 above)
Comment (xhoaluu) [inline sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java]: adding comment with link issue, RFC 3261 reference link and brief explanation.

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/concurrency/ConcurrentyControlSipSessionIsolationTest.java (hunk 2 above)
Comment (xhoaluu) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/concurrency/ConcurrentyControlSipSessionIsolationTest.java]: No, we don't need to modify this existing testsuite, that was my mistake when adding new  testcase.

Removed the modification.

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java (hunk 4 above)
Comment (xhoaluu) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java]: I'm expecting 481 Call/Transaction does not exist respond to INFO request. it's not for initial INVITE.

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java (hunk 3 above)
Comment (xhoaluu) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java]: yes, the first INVITE is not initial invite. I add totag in to the request TO header to make it to be an in-dialog and accept it , this is the trick from me to make the sender possible to send subsequence INFO and BYE request. then the BYE also for BYE a non-existing dialog.

File: sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java (hunk 4 above)
Comment (jaimecasero) [inline sip-servlets-test-suite/testsuite/src/test/java/org/mobicents/servlet/sip/testsuite/simple/ShootmeSipServletTest.java]: ok

File: sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java [hunk 5]
```diff
@@ -190,11 +190,15 @@ public void dispatchMessage(final SipProvider sipProvider, SipServletMessageImpl
 							}				
 							return ;
 						} else {
+						    // Fixed github issue: https://github.com/RestComm/sip-servlets/issues/189
+						    // RFC 3261: https://tools.ietf.org/html/rfc3261#section-12.2.2
```
Comment (jaimecasero) [inline sip-servlets-impl/src/main/java/org/mobicents/servlet/sip/core/dispatchers/SubsequentRequestDispatcher.java]: i like this comment, please get accostume to write this kind of good stuff into your changes

**Comment by jaimecasero** [comment]
closing this one as discussed to start with fresh one

**Pull request closed without merging**